03week/towersOfHanoi.js

```diff
@@ -19,24 +19,39 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  const block = stacks[startStack].pop();
+  stacks[endStack].push(block);
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+
+  const startIndex = startStack.length - 1;
+  const startBlock = stacks[startStack][startIndex];
+  const endIndex = endStack.length - 1;
+  const endBlock = stacks[endStack][endIndex];
+
+
+  if (stacks[endStack].length === 0 || startIndex < endIndex) {
```

Comment:
I think you meant to compare the blocks at `startIndex` and `endIndex` instead of comparing the indices themselves